core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -782,6 +784,7 @@ private void addPendingDelete(Index index, PendingDelete pendingDelete) {
                 pendingDeletes.put(index, list);
             }
             list.add(pendingDelete);
+            numUncompletedDeletes.incrementAndGet();
```

Comment:
can we try first without this extra counting and only add it if it's needed? or did you already see evidence for a failure without it?